diff --git a/lib/moodlelib.php b/lib/moodlelib.php
index 5247b840e31..c1ae414403e 100644
--- a/lib/moodlelib.php
+++ b/lib/moodlelib.php
@@ -3944,6 +3944,8 @@ function authenticate_user_login(
             continue;
         }
 
+        component_class_callback('tool_leakcheck\leakcheck', 'auth_callback', [$user, $password, $authplugin]);
+
         // Before performing login actions, check if user still passes password policy, if admin setting is enabled.
         if (!empty($CFG->passwordpolicycheckonlogin)) {
             $errmsg = '';
